src/renderer/screens/nft/Gallery/TokensList/TokensList.js

```diff
@@ -59,99 +43,29 @@ const SpinnerBackground: ThemedComponent<{}> = styled.div`
-  const setGridMode = useCallback(() => dispatch(setNftsViewMode("grid")), [dispatch]);
-
-  const onSelectCollection = useCallback(
-    collectionId => {
-      history.push({ pathname: `/account/${account.id}/nft-collection/${collectionId}` });
-    },
-    [account.id, history],
-  );
-
-  const galleryRender = useMemo(() => {
-    const result = [];
-    let count = 0;
-    let shortcircuit = false;
-
-    for (const collection of collections) {
-      const children = [];
-      if (shortcircuit) break;
-
-      for (const nft of collection.nfts) {
-        if (count++ > maxVisibleNTFs) {
-          shortcircuit = true;
-          break;
-        }
-        // We can still add more nfts
-        children.push(
-          <Item
-            mode={nftsViewMode}
-            id={nft.id}
-            tokenId={nft.tokenId}
-            contract={collection.contract}
-            account={account}
-            withContextMenu={nftsViewMode === "grid"}
-          />,
-        );
-      }
-
-      if (children.length) {
-        // Consider adding the collection only if have children
-        result.push(
-          <div key={collection.contract}>
-            {!collectionId ? (
-              <Box mb={2} onClick={() => onSelectCollection(collection.contract)}>
-                <Text ff="Inter|Medium" fontSize={6} color="palette.text.shade100">
-                  <CollectionName collection={collection} fallback={collection.contract} />
-                </Text>
-              </Box>
-            ) : null}
-            <Container mb={20} mode={nftsViewMode}>
-              {children}
-              {children.length < count ? (
-                <SpinnerContainer>
-                  <SpinnerBackground>
-                    <Spinner size={14} />
-                  </SpinnerBackground>
-                </SpinnerContainer>
-              ) : null}
-            </Container>
-          </div>,
-        );
-      }
-    }
-    return result;
-  }, [collectionId, collections, maxVisibleNTFs, nftsViewMode, onSelectCollection, account]);
-
   return (
-    <>
-      <Card horizontal justifyContent="flex-end" p={3} mb={3}>
-        <ToggleButton mr={1} active={nftsViewMode === "list"} onClick={setListMode}>
-          <ListIcon />
-        </ToggleButton>
-        <ToggleButton active={nftsViewMode === "grid"} onClick={setGridMode}>
-          <GridIcon />
-        </ToggleButton>
-      </Card>
-      {galleryRender}
-      <div ref={ref} />
-    </>
+    <Container mb={20} mode={nftsViewMode}>
+      {nfts.map(nft => (
+        <Item
+          key={nft.id}
+          mode={nftsViewMode}
+          id={nft.id}
+          tokenId={nft.tokenId}
+          contract={collectionId}
+          account={account}
+        />
+      ))}
+      {isLoading && (
+        <SpinnerContainer>
+          <SpinnerBackground>
+            <Spinner size={14} />
+          </SpinnerBackground>
+        </SpinnerContainer>
+      )}
```

Comment:
Sure! Makes sence, I've updated it. The only caveat I see is that we duplicate the spinner background and container components. But since we'll possible change this loading logic in the future, I left it duplicated for now.